### PR TITLE
I48 api keys

### PR DESCRIPTION
**Pull request by hajnalandor**

### CODE REVIEW

File: pkg/client/http/client.go [hunk 1]
```diff
@@ -21,24 +21,31 @@ type Options struct {
 	*api.Config
 	EngineName string
 	URLs       []string
+	Token []string
```
Comment (borosr) [inline pkg/client/http/client.go:24]: This should be a single token for each heimdall nodes.

**Review by borosr — CHANGES_REQUESTED** [review]

File: cmd/flags/flags.go [hunk 2]
```diff
@@ -101,4 +102,8 @@ var (
 		Usage: "",
 		Value: false,
 	}
+	TokenId = &cli.StringFlag{
```
Comment (borosr) [inline cmd/flags/flags.go:105]: We should add it into server cmd to be able to pre-set API key in development mode (--in-memory). Skip it in production mode (if backend is set).

**Review by borosr — CHANGES_REQUESTED** [review]

**Review by borosr — APPROVED** [review]

File: pkg/client/http/client.go [hunk 3]
```diff
@@ -21,14 +21,22 @@ type Options struct {
...
 	EngineName string
 	URLs       []string
+	Token      string
 }
 
 func (o *Options) Setup() client.Client {
 	if o.Config == nil {
 		o.Config = api.DefaultConfig()
 	}
+	if len(o.URLs) != len(o.Token) {
```
Comment (borosr) [inline pkg/client/http/client.go:31]: Remove this please.

**Review by borosr — CHANGES_REQUESTED** [review]

**Review by PumpkinSeed — CHANGES_REQUESTED** [review]
- What happens if I don't pass `--id` in-memory mode and server?
- Use Vault's root token
- New token should require previous available root-token
- GRPC use token as well
- Rename id flag

**Comment by borosr** [comment]
Heimdall can use vault's root and other tokens.

**Review by PumpkinSeed — APPROVED** [review]

**Pull request merged**